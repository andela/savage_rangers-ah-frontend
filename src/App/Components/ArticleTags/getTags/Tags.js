import React, { Component } from 'react';
import axios from 'axios';
import Tag from './Tag';

export default class Tags extends Component {
  state = { tags: [] };

  componentWillMount() {
    axios
      .get('https://authors-heaven.herokuapp.com/api/articles/How-to-create-sequalize-seeds/tags')
      .then(res => this.setState({ tags: res.data.data }));
  }

  render() {
    const getTags = this.state.tags.map(tag => <Tag key={tag.id} body={tag.name} />);
    return <div>{getTags}</div>;
  }
}
