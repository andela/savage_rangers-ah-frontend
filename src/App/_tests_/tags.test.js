import React from 'react';
import { shallow } from '../../enzyme';
import Tags from '../Components/DisplayArticle/tags';

const data = [
  {
    id: 4,
    name: 'Software',
    createdAt: '2019-08-14T12:56:55.150Z',
    updatedAt: '2019-08-14T12:56:55.150Z'
  },
  {
    id: 6,
    name: 'Growing',
    createdAt: '2019-08-14T12:56:55.152Z',
    updatedAt: '2019-08-14T12:56:55.152Z'
  }
];
describe('testing the tags component', () => {
  const tags = shallow(<Tags tags={data} />);
  it('should test if the component renders the appropriate jsx elements', () => {
    expect(tags.find('div').exists()).toBe(true);
    expect(tags).toMatchSnapshot();
  });
});
