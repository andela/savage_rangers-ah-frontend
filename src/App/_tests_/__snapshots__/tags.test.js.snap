// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the tags component should test if the component renders the appropriate jsx elements 1`] = `ShallowWrapper {}`;
